helix-core/src/test/java/org/apache/helix/controller/rebalancer/waged/MockAssignmentMetadataStore.java

```diff
@@ -41,17 +41,19 @@
     return _persistGlobalBaseline;
   }
 
-  public void persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
+  public boolean persistBaseline(Map<String, ResourceAssignment> globalBaseline) {
     _persistGlobalBaseline = globalBaseline;
+    return true;
```

Comment:
This behavior might be confusing. This is not ideal... usually people would think if I get a false, that means that a write to metadata store failed (for example, ZK write failure). But here, that's not necessarily the case. In this case, we decide not to write it on purpose because there's no diff.

How should we differentiate this behavior clearly? No diff != Write failure and should make each case be expressed distinctly?

Reply:
Let's keep the same discussion in the same comment.